src/screens/SendToken/SendTokenAmount.js

```diff
@@ -187,57 +204,84 @@ class SendTokenAmount extends React.Component<Props, State> {
   };
 
   handleChange = (value: Object) => {
-    this.setState({ value });
-    this.updateGasLimit();
+    // first update the amount, then after state is updated check for errors
+    this.setState({ value, gettingFee: true });
+    this.getTxFeeInWei()
+      .then(txFeeInWei => this.setState({ txFeeInWei, gettingFee: false }))
+      .catch(() => null);
+    this.checkFormInputErrors();
+    if (!checkIfSmartWalletAccount(this.props.activeAccount)) {
+      this.updateGasLimit();
+    }
   };
 
-  handleFormSubmit = () => {
+  handleFormSubmit = async () => {
+    const { submitPressed } = this.state;
+    if (submitPressed) return;
     this.formSubmitted = true;
-    const txFeeInWei = this.getTxFeeInWei();
+    this.setState({ submitPressed: true });
+    const txFeeInWei = await this.getTxFeeInWei();
```

Comment:
Don't we have this on the state already?

Reply:
We want to get it according latest price this is why we were doing the same with previous calculation. It can may change every second and holding user for 1s on this purpose is fine.